Add compatibitily matrix (#4874)
realm/realm-js: README.md

## Getting Started

> Please see the detailed instructions in our docs to use [Realm JavaScript for Node.js](https://docs.mongodb.com/realm/sdk/node/) and [Realm JavaScript for React Native](https://docs.mongodb.com/realm/sdk/react-native/). Please notice that currently only Node.js version 13 or later are supported. Moreover, you might want to check out our [compatibility matrix](COMPATIBILITY.md).

Reviewer: ```suggestion
Please see the detailed instructions in our docs to use [Realm JavaScript for Node.js](https://docs.mongodb.com/realm/sdk/node/) and [Realm JavaScript for React Native](https://docs.mongodb.com/realm/sdk/react-native/). Please notice that currently only Node.js version 13 is later are supported. For React Native users, we have a [compatibility matrix](COMPATIBILITY.md) showing which versions are supported.
```
